test/js/specs/unit/views/EditorView.spec.js

```diff
@@ -0,0 +1,370 @@
+        expect(state.accessPolicy.length).to.equal(3); // 2 initial + 1 additional
+
+        // Mock the save functionality
+        var saveSpy = sinon.spy();
+        state.dataObject.save = saveSpy;
+
+        // Simulate save (would normally be triggered by UI)
+        if (state.editorView.save) {
+          state.editorView.save();
+        } else {
+          // Directly call save on the model
+          state.dataObject.save();
+        }
+
+        // Verify access policy is preserved
+        var policyAfterSave = state.dataObject.get("accessPolicy");
+        expect(policyAfterSave).to.exist;
+        expect(policyAfterSave).to.equal(state.accessPolicy);
+        expect(policyAfterSave.length).to.equal(3);
+
+        // Verify all subjects are preserved
+        var subjects = policyAfterSave.pluck("subject").sort();
+        expect(subjects).to.deep.equal([
+          "public",
+          "uid=collaborator,dc=example", 
+          "uid=testuser,dc=example"
+        ]);
+      });
+    });
+
+    describe("AccessPolicy Error Handling", function() {
+      beforeEach(function() {
+        state.editorView = new EditorView({
+          model: state.dataObject
+        });
+      });
+
+      it("should handle missing AccessPolicy gracefully", function() {
+        // Create a DataONEObject without explicitly setting an accessPolicy
+        var testObject = new DataONEObject({
+          id: "test-no-policy",
+          type: "Data"
+        });
+        
+        var policy = testObject.get("accessPolicy");
+        
+        // The policy might be undefined or an empty collection
+        if (policy) {
+          expect(policy.length).to.equal(0);
+        } else {
+          expect(policy).to.be.undefined;
+        }
+
+        // The editor should handle this gracefully
+        expect(function() {
+          var editorView = new EditorView({ model: testObject });
+          if (editorView.renderAccessPolicy) {
+            // This should not throw even with missing policy
+            // (it will exit early due to no allowAccessPolicyChanges)
+          }
+        }).to.not.throw();
+      });
+
+      it.skip("EXPECTED FAILURE: should handle roundtrip AccessPolicy data", function() {
+        // Create AccessPolicy with sample data
+        var policyData = new AccessPolicy();
+        policyData.add([
+          { subject: "uid=test,dc=example", read: true },
+          { subject: "public", read: true }
+        ]);
+        const rawPolicyData = policyData.toJSON();
+        
+        state.dataObject.set("accessPolicy", rawPolicyData);
+
+        // Verify it's raw data (this is the serialization bug)
+        var policy = state.dataObject.get("accessPolicy");
+        expect(Array.isArray(policy)).to.be.true;
+
+        // Now simulate what happens during normal reload - let automatic deserialization fail
+        var jsonData = state.dataObject.toJSON();
+        var reloadedObject = new DataONEObject(jsonData);
```

Comment:
Similar questions here, why to JSON and when/how is the dataone obj data being reloaded in the app?